lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -627,18 +630,76 @@ export default function getAPIImplementation(): APIInterface {
                 userContext,
             });
 
-            let session = await Session.createNewSession(
-                options.req,
-                options.res,
+            const mfaInstance = MultiFactorAuthRecipe.getInstance();
+
+            if (mfaInstance === undefined) {
+                // No MFA stuff here, so we just create and return the session
+                let session = await Session.getSession(
+                    options.req,
+                    options.res,
+                    { sessionRequired: false },
+                    userContext
+                );
+
+                session = await Session.createNewOrKeepExistingSession(
+                    options.req,
+                    options.res,
+                    tenantId,
+                    emailPasswordRecipeUser.recipeUserId,
+                    {},
+                    {},
+                    userContext
+                );
+
+                return {
+                    status: "OK",
+                    session,
+                    user: response.user,
+                };
+            }
+
+            let session: SessionContainerInterface | undefined;
+            session = await Session.getSession(options.req, options.res, { sessionRequired: false });
```

Comment:
When loading sessions like this we should use `overrideGlobalClaimValidators: () => []`, otherwise these will throw a 304 if the user hasn't finished signing in (even if this is the thing they need to finish sign in)

Reply:
Done